test/powershell/engine/Api/GetMethodInfoOverloadDefinition.Tests.ps1

```diff
@@ -0,0 +1,166 @@
+
+    Context "Verify Definition of Parameterized Property" {
+        BeforeAll {
+            $classDefinition1 = @"
+namespace ParameterizedPropertyDefinitionTest
+{
+    public class TestClass
+    {
+        private string[] _indexerArray = new string[1];
+        public string this[int i]
+        {
+            get => _indexerArray[i];
+            set => _indexerArray[i] = value;
+        }
+    }
+}
+"@
+            $classDefinition2 = @"
+namespace ReadonlyParameterizedPropertyDefinitionTest
+{
+    public class TestClass
+    {
+        public int this[int i] => 42;
+    }
+}
+"@
+            $classDefinition3 = @"
+namespace ReadonlyParameterizedPropertyWithChangedNameDefinitionTest
+{
+    public class TestClass
+    {
+        [System.Runtime.CompilerServices.IndexerName("TheItem")]
+        public int this[int i] => 42;
+    }
+}
+"@
+            $TestType1 = Add-Type -PassThru -TypeDefinition $classDefinition1
+            $TestType2 = Add-Type -PassThru -TypeDefinition $classDefinition2
+            $TestType3 = Add-Type -PassThru -TypeDefinition $classDefinition3
+        }
+
+        It "Get definition of parametrized property" {
+            $obj = $TestType1::new()
+            $result = $obj | Get-Member -Type ParameterizedProperty
+            $result.Definition | Should -BeExactly "string Item(int i) {get;set;}"
+        }
+
+        It "Get definition of readonly parametrized property" {
+            $obj = $TestType2::new()
+            $result = $obj | Get-Member -Type ParameterizedProperty
+            $result.Definition | Should -BeExactly "int Item(int i) {get;}"
+        }
+
+        It "Get definition of parametrized property with changed name" {
+            $obj = $TestType3::new()
+            $result = $obj | Get-Member -Type ParameterizedProperty
+            $result.Name | Should -BeExactly "TheItem"
+            $result.Definition | Should -BeExactly "int TheItem(int i) {get;}"
+        }
+    }
+
+    Context "Verify overloaded method definitions" {
+        BeforeAll {
+            Add-Type -NameSpace OverloadDefinitionTest -Name TestClass -MemberDefinition @"
+public static void Bar() { }
+public static void Bar(int param) { }
+public static void Bar(int param1, string param2) { }
+public static string Bar(int? param1, string param2) { return param2; }
+"@
+        }
+
+        It "Get methods' overload definitions" {
+            $definitions = ([OverloadDefinitionTest.TestClass]::Bar).OverloadDefinitions
+            $definitions.Count | Should -BeExactly 4
+            $definitions[0] | Should -BeExactly "static void Bar()"
+            $definitions[1] | Should -BeExactly "static void Bar(int param)"
+            $definitions[2] | Should -BeExactly "static void Bar(int param1, string param2)"
+            $definitions[3] | Should -BeExactly "static string Bar(System.Nullable[int] param1, string param2)"
+        }
+    }
+
+    Context "Verify method definitions with different parameter set" {
```

Comment:
"parameter set" confuses me. Do you mean "method signature"?
Also the description should more exactly describe intention/code path of the test (otherwise it looks like previous test).

Reply:
> You can use TabCompletion2 (see our tests how it is used).

I couldn't find TabCompletion2 anywhere. Do you mean TabExpansion2? Anyway, as far as I understand, tab completion completes just method name and there are already tests for that.

> Do you mean "method signature"?

Rather, "method signature with parameters of various types such as parameters passed by reference, generic and optional parameters", but it seems to me, we could combine tests for the overloaded definition and different method signature into one context.

I also want to suggest to mark parameters passed by reference with `[in]` and `[out]` keywords in addition to the `[ref]`. Can I implement that in this PR?